src/main/kotlin/app/revanced/patcher/apk/ApkBundle.kt

```diff
@@ -0,0 +1,156 @@
+package app.revanced.patcher.apk
+
+import app.revanced.patcher.Patcher
+import app.revanced.patcher.PatcherOptions
+import com.reandroid.apk.ApkModule
+import com.reandroid.apk.ResourceIds
+import com.reandroid.apk.xmlencoder.EncodeMaterials
+import com.reandroid.arsc.util.FrameworkTable
+import com.reandroid.arsc.value.ResConfig
+import com.reandroid.common.TableEntryStore
+import java.io.File
+
+/**
+ * An [Apk] file of type [Apk.Split].
+ *
+ * @param files A list of apk files to load.
+ */
+class ApkBundle(files: List<File>) {
+    /**
+     * The [Apk.Base] of this [ApkBundle].
+     */
+    val base: Apk.Base
+
+    /**
+     * A map containing all the [Apk.Split]s in this bundle associated by their configuration.
+     */
+    val splits: Map<String, Apk.Split>?
+
+    private fun ApkModule.isFeatureModule() = androidManifestBlock.manifestElement.let {
+        it.searchAttributeByName("isFeatureSplit")?.valueAsBoolean == true || it.searchAttributeByName("configForSplit") != null
+    }
+
+    init {
+        var baseApk: Apk.Base? = null
+        val splitList = mutableListOf<Apk.Split>()
+
+        files.forEach {
+            val module = ApkModule.loadApkFile(it)
+            when {
+                module.isBaseModule -> {
+                    if (baseApk != null) {
+                        throw IllegalArgumentException("Cannot have more than one base apk")
+                    }
+                    baseApk = Apk.Base(module)
+                }
+
+                !module.isFeatureModule() -> {
+                    val config = module.split.removePrefix("config.")
+
+                    splitList.add(
+                        when {
+                            config.length == 2 -> Apk.Split.Language(config, module)
+                            Apk.Split.Library.architectures.contains(config) -> Apk.Split.Library(config, module)
+                            ResConfig.Density.valueOf(config) != null -> Apk.Split.Asset(config, module)
+                            else -> throw IllegalArgumentException("Unknown split: $config")
+                        }
```

Comment:
These conditions look very arbitrary to me. Why does config.length == 2 result to a language apk, or ResConfig.Density.valueOf(config) != null to Asset apk? Please document the code and explain such things.

Reply:
Alright. I added comments